### PR TITLE
Upgrade to zcbor 0.6.0

### PR DESCRIPTION
**Pull request by oyvindronningstad**

### CODE REVIEW

**Comment by oyvindronningstad** [comment]
cc @VeijoPesonen

**Review by rlubos — DISMISSED** [review]

File: subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c [hunk 1]
```diff
@@ -49,14 +47,13 @@ static bool encode_repeated_record_bt(zcbor_state_t *state, const struct record_
 	zcbor_print("%s\r\n", __func__);
 
 	bool tmp_result = ((((zcbor_int32_put(state, (-3)))) &&
-			    ((((*input)._record_bt >= INT64_MIN) &&
-			      ((*input)._record_bt <= INT64_MAX)) ||
+			    ((((*input)._record_bt >= -9223372036854775807LL) &&
+			      ((*input)._record_bt <= 9223372036854775807LL)) ||
```
Comment (nordicjm) [inline subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c:51]: Any reason for the switch from defines to literals?

File: subsys/net/lib/lwm2m/lwm2m_senml_cbor_types.h [hunk 2]
```diff
@@ -21,9 +21,7 @@
 
 enum lwm2m_senml_cbor_key {
 	lwm2m_senml_cbor_key_bn = -2,
-	lwm2m_senml_cbor_key_bt = -3,
```
Comment (VeijoPesonen) [inline subsys/net/lib/lwm2m/lwm2m_senml_cbor_types.h:24]: Both keys should be put back.

**Review by VeijoPesonen — CHANGES_REQUESTED** [review]
Array size calculation coming from ZCBOR side is a nice addition.

**Comment by VeijoPesonen** [comment]
@jheiskan81 Please review.

File: subsys/net/lib/lwm2m/lwm2m_senml_cbor_decode.c [hunk 3]
```diff
@@ -34,9 +36,24 @@ static bool decode_repeated_record_bn(zcbor_state_t *state, struct record_bn *re
...
+}
+
+static bool decode_repeated_record_bt(zcbor_state_t *state, struct record_bt *result)
+{
+	zcbor_print("%s\r\n", __func__);
+
+	bool tmp_result = ((((zcbor_int32_expect(state, (-3)))) &&
+			    (zcbor_int64_decode(state, (&(*result)._record_bt))) &&
+			    ((((*result)._record_bt >= -9223372036854775807LL) &&
+			      ((*result)._record_bt <= 9223372036854775807LL)) ||
```
Comment (oyvindronningstad) [inline subsys/net/lib/lwm2m/lwm2m_senml_cbor_decode.c:52]:
```suggestion
			    ((((*result)._record_bt >= INT64_MIN) &&
			      ((*result)._record_bt <= INT64_MAX)) ||
```

File: subsys/net/lib/lwm2m/lwm2m_senml_cbor_decode.c [hunk 4]
```diff
@@ -48,9 +65,24 @@ static bool decode_repeated_record_n(zcbor_state_t *state, struct record_n *resu
...
+}
+
+static bool decode_repeated_record_t(zcbor_state_t *state, struct record_t *result)
+{
+	zcbor_print("%s\r\n", __func__);
+
+	bool tmp_result = ((((zcbor_uint32_expect(state, (6)))) &&
+			    (zcbor_int64_decode(state, (&(*result)._record_t))) &&
+			    ((((*result)._record_t >= -9223372036854775807LL) &&
+			      ((*result)._record_t <= 9223372036854775807LL)) ||
```
[the remark this answers is not in the document]
Comment (oyvindronningstad) [reply subsys/net/lib/lwm2m/lwm2m_senml_cbor_decode.c:81]:
```suggestion
			    ((((*result)._record_t >= INT64_MIN) &&
			      ((*result)._record_t <= INT64_MAX)) ||
```

File: subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c (hunk 1 above)
Comment (oyvindronningstad) [inline subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c:51]:
```suggestion
			    ((((*input)._record_bt >= INT64_MIN) &&
			      ((*input)._record_bt <= INT64_MAX)) ||
```

File: subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c [hunk 5]
```diff
@@ -79,15 +75,14 @@ static bool encode_repeated_record_t(zcbor_state_t *state, const struct record_t
...
 	zcbor_print("%s\r\n", __func__);
 
-	bool tmp_result =
-		((((zcbor_uint32_put(state, (6)))) &&
-		  ((((*input)._record_t >= INT64_MIN) && ((*input)._record_t <= INT64_MAX)) ||
-		   (zcbor_error(state, ZCBOR_ERR_WRONG_RANGE), false)) &&
-		  (zcbor_int64_encode(state, (&(*input)._record_t)))));
+	bool tmp_result = ((((zcbor_uint32_put(state, (6)))) &&
+			    ((((*input)._record_t >= -9223372036854775807LL) &&
+			      ((*input)._record_t <= 9223372036854775807LL)) ||
```
[the remark this answers is not in the document]
Comment (oyvindronningstad) [reply subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c:80]:
```suggestion
			    ((((*input)._record_t >= INT64_MIN) &&
			      ((*input)._record_t <= INT64_MAX)) ||
```

File: subsys/net/lib/lwm2m/lwm2m_senml_cbor_types.h [hunk 6]
```diff
@@ -21,9 +21,7 @@
 
 enum lwm2m_senml_cbor_key {
 	lwm2m_senml_cbor_key_bn = -2,
```
Comment (oyvindronningstad) [inline subsys/net/lib/lwm2m/lwm2m_senml_cbor_types.h:23]:
```suggestion
	lwm2m_senml_cbor_key_bn = -2,
	lwm2m_senml_cbor_key_bt = -3,
```

File: subsys/net/lib/lwm2m/lwm2m_senml_cbor_types.h [hunk 7]
```diff
@@ -21,9 +21,7 @@
...
-	lwm2m_senml_cbor_key_bt = -3,
 	lwm2m_senml_cbor_key_n  = 0,
```
Comment (oyvindronningstad) [inline subsys/net/lib/lwm2m/lwm2m_senml_cbor_types.h:24]:
```suggestion
	lwm2m_senml_cbor_key_n  = 0,
	lwm2m_senml_cbor_key_t  = 6,
```

File: subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c [hunk 8]
```diff
@@ -139,8 +133,10 @@ static bool encode_value(zcbor_state_t *state, const struct value_ *input)
 			: (((*input)._value_choice == _value_bstr)
 				   ? ((zcbor_bstr_encode(state, (&(*input)._value_bstr))))
 				   : (((*input)._value_choice == _value_int)
-					      ? (((((*input)._value_int >= INT64_MIN) &&
-						   ((*input)._value_int <= INT64_MAX)) ||
+					      ? (((((*input)._value_int >=
+						    INT64_MIN) &&
+						   ((*input)._value_int <=
+						    INT64_MAX)) ||
```
Comment (oyvindronningstad) [inline subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c:139]:
```suggestion
					      ? (((((*input)._value_int >= INT64_MIN) &&
						   ((*input)._value_int <= INT64_MAX)) ||
```

**Comment by rlubos** [comment]
@oyvindronningstad Out of curiosity, I was under the impression that this code was autogenerated and there's not much room for modifications?

**Comment by oyvindronningstad** [comment]
@rlubos The generated code is formatted with clang-format, and then a patch is applied.

File: subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c [hunk 9]
```diff
@@ -37,9 +36,8 @@ static bool encode_repeated_record_bn(zcbor_state_t *state, const struct record_
 	bool tmp_result = ((((zcbor_int32_put(state, (-2)))) &&
 			    (zcbor_tstr_encode(state, (&(*input)._record_bn)))));
 
-	if (!tmp_result) {
```
Comment (jheiskan81) [inline subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c:40]: Is this clang rules different than normal zephyr. I was wondering that {
} is mandatory.

Comment (oyvindronningstad) [reply subsys/net/lib/lwm2m/lwm2m_senml_cbor_encode.c:40]: I think the clang-format automatically uses the zephyr rules.

**Review by VeijoPesonen — DISMISSED** [review]
Looks good to me.

File: doc/releases/release-notes-3.3.rst [hunk 10]
```diff
@@ -261,6 +265,19 @@ MCUboot
...
 ******************
 
+zcbor
+*****
+
+Upgraded zcbor to 0.6.0. Among other things, this brings in a few convenient
+changes for Zephyr:
+
+* In the zcbor codebase, the ``ARRAY_SIZE`` macro has been renamed to
+  ``ZCBOR_ARRAY_SIZE`` to not collide with Zephyr's ``ARRAY_SIZE_`` macro.
```
Comment (nordicjm) [inline doc/releases/release-notes-3.3.rst:275]:
```suggestion
  ``ZCBOR_ARRAY_SIZE`` to not collide with Zephyr's ``ARRAY_SIZE`` macro.
```

File: west.yml [hunk 11]
```diff
@@ -253,7 +253,7 @@ manifest:
       revision: e8920192b66db4f909eb9cd3f155d5245c1ae825
       path: modules/lib/uoscore-uedhoc
     - name: zcbor
-      revision: a0d6981f14d4001d6f0d608d1a427f9bc6bb6d02
+      revision: 0.6.0
```
Comment (nordicjm) [inline west.yml:256]: Should this not be a commit ID?

Comment (de-nordic) [reply west.yml:256]: We have tags now on zcbor fork. The tags are the same as in upstream.

File: doc/releases/release-notes-3.3.rst (hunk 10 above)
Comment (oyvindronningstad) [reply doc/releases/release-notes-3.3.rst]: Changed to
```
:c:macro:`ARRAY_SIZE`
```

**Review by nordicjm — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Comment by de-nordic** [comment]
@carlescufi You gave me 1minute to approve ;). I know that I am to slow with reviews, but at least give me a chance :D.
Just kidding: I have already approved yesterday, and there were no changes.

**Pull request merged**